.github/workflows/ai-generated.yml

```diff
@@ -0,0 +1,58 @@
+---
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+name: 'AI-generated pull requests'
+
+on:
+  pull_request:
+    types: [labeled, unlabeled, reopened]
+
+permissions:
+  pull-requests: read
+
+jobs:
+  ai-generated-support:
+    if: github.event.label.name == 'ai-generated' || github.event.action == 'reopened'
+    runs-on: ubuntu-24.04
+    concurrency:
+      group: support-${{ github.event.pull_request.number }}
+      cancel-in-progress: true
+    permissions:
+      pull-requests: write
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GH_REPO: ${{ github.repository }}
+      NUMBER: ${{ github.event.pull_request.number }}
+      PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+    steps:
+      - name: Label added, comment and close pull request
+        if: github.event.action == 'labeled' && github.event.label.name == 'ai-generated'
+        shell: bash
+        env:
+          BODY: >
+            :wave: @${{ env.PR_AUTHOR }}, thank you for your contribution!
```

Comment:
Perhaps we should also add a paragraph noting that this pull request is too large for effective human review. The author should discuss with us on how to break down these changes into smaller, more focused PRs to ensure a thorough and efficient review process.

Reply:
Yep, nice suggestion